core/plugin_api.rs

```diff
@@ -20,4 +23,5 @@ pub type DispatchOpFn =
 
 pub trait Interface {
   fn register_op(&mut self, name: &str, dispatcher: DispatchOpFn) -> OpId;
+  fn resource_table(&mut self) -> Rc<RefCell<ResourceTable>>;
```

Comment:
Without wrapping the resource table, we can't ensure that the plugin stays loaded until resources it creates are dropped. This will result in segfaults.

Reply:
The plugin will be loaded until the isolate is dropped due to the op dispatcher wrapper.

Reply:
If this is the case then leaving a resource in the resource table at the end of the tests shouldn't cause a segfault when exiting. Here's a patch:

``` patch
diff --git a/test_plugin/src/lib.rs b/test_plugin/src/lib.rs
index 0f509e60..87446014 100644
--- a/test_plugin/src/lib.rs
+++ b/test_plugin/src/lib.rs
@@ -118,5 +118,14 @@ fn op_test_resources(
     let found2 = rt.close(rid).is_some();
     assert!(!found2);
   }
+  {
+    // add and leave in table
+    let rc = Box::new(TestResource {
+      noise: "woof".to_owned(),
+    });
+    let rt = interface.resource_table();
+    let mut rt = rt.borrow_mut();
+    rt.add("test_resource", rc);
+  }
   Op::Sync(Default::default())
 }
```

Reply:
@afinch7 Agree - I've added your patch